include/seqan3/range/detail/random_access_iterator.hpp

```diff
@@ -0,0 +1,259 @@
+    bool operator!=(random_access_iterator const & rhs) const
+    {
+        return pos != rhs.pos;
+    }
+
+    //! \brief Iterator comparison refers to their positions
+    bool operator<(random_access_iterator const & rhs) const
+    {
+        return static_cast<bool>(pos < rhs.pos);
+    }
+
+    //! \brief Iterator comparison refers to their positions
+    bool operator>(random_access_iterator const & rhs) const
+    {
+        return pos > rhs.pos;
+    }
+
+    //! \brief Iterator comparison refers to their positions
+    bool operator<=(random_access_iterator const & rhs) const
+    {
+        return pos <= rhs.pos;
+    }
+
+    //! \brief Iterator comparison refers to their positions
+    bool operator>=(random_access_iterator const & rhs) const
+    {
+        return pos >= rhs.pos;
+    }
+    //!\}
+
+    /*!\name Arithmetic operators
+     * \{
+    */
+    //! \brief Pre-increment, return updated iterator
+    random_access_iterator & operator++()
+    {
+        ++pos;
+        return (*this);
+    }
+
+    //! \brief Post-increment, return previous iterator state
+    random_access_iterator operator++(int)
+    {
+        random_access_iterator cpy{*this};
+        ++pos;
+        return cpy;
+    }
+
+    //! \brief Pre-decrement, return updated iterator
+    random_access_iterator& operator--()
+    {
+        --pos;
+        return *this;
+    }
+
+    //! \brief Post-decrement, return previous iterator state
+    random_access_iterator operator--(int)
+    {
+        random_access_iterator cpy{*this};
+        --pos;
+        return cpy;
+    }
+
+    //! \brief Forward this iterator
+    random_access_iterator& operator+=(const difference_type skip)
+    {
+        pos += skip;
+        return *this;
+    }
+
+    //! \brief Forward copy of this iterator
+    random_access_iterator operator+(const difference_type skip) const
+    {
+        return random_access_iterator{*host, static_cast<position_type>(pos + skip)};
+    }
+
+    //! \brief Non-member operator+ delegates to non-friend operator+
+    friend random_access_iterator operator+(const difference_type skip , const random_access_iterator& it)
+    {
+        return it + skip;
+    }
+
+    //! \brief Decrement iterator by skip
+    random_access_iterator& operator-=(const difference_type skip)
+    {
+        pos -= skip;
+        return *this;
+    }
+
+    //! \brief Return decremented copy of this iterator
+    random_access_iterator operator-(const difference_type skip) const
+    {
+        return random_access_iterator{*host, static_cast<position_type>(pos - skip)};
+    }
+
+    //! \brief Non-member operator- delegates to non-friend operator-
+    friend random_access_iterator operator-(const difference_type skip , const random_access_iterator& it)
+    {
+        return it - skip;
+    }
+
+    //! \brief Return offset between this and remote iterator's position
+    difference_type operator-(random_access_iterator lhs) const
```

Comment:
Should it return a new iterator (would be intuitive), i.e. same behaviour than operator+(skip)? Then why does the difference it1-it2 returns only an offset and not an iterator? Would be good to have a consistent behaviour. What do you think?